src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2470,6 +2470,24 @@ static bool contextElementPreconditionsCheck
   /* Getting the entity in the request (helpful in other places) */
   EntityId* enP = &ceP->entityId;
 
+  /* Checking there aren't duplicate attributes */
+  for (unsigned int ix = 0; ix < ceP->contextAttributeVector.size(); ++ix)
+  {
+    std::string name = ceP->contextAttributeVector.get(ix)->name;
+    std::string id   = ceP->contextAttributeVector.get(ix)->getId();
+    for (unsigned int jx = ix + 1; jx < ceP->contextAttributeVector.size(); ++jx)
```

Comment:
Note the initial value in the loop, starting in the position of the attribute being compared for duplicates. I understand that this optimizes the loop (as we can save many items) as we have "passed" cases already evaluated for duplication, but a second opinion would be welcome.